### DISCUSSION TITLE
Not sending response

### DISCUSSION THREAD

**Discussion by iv-med** [jaredhanson/passport#821] [Help]
Hi everybody!
I have a problem when I want to send response to frontend: 

"Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client"

```
passport.use(new LocalStrategy(
  function(username, password, done) {
    console.log(username);
 users.map((u) => {
  if((u.login === username) && (u.password === password)){
    return done(null, username);
  }
 });
 return done(null, false);   
  }
));

app.post('/api/getAuth', passport.authenticate('local', { failureRedirect: '/login' }),
function(req, res) {
  return res.sendStatus(200);
});
```
In frontend log I see net::ERR_EMPTY_RESPONSE

passport v0.4.0, express v4.17.1

**Comment by ghost** [comment]
Looks like a fairly standard you've already sent your response error. But from the looks of it your composition to return the call of each user, you might be slightly off track. I thought maybe, okay they are testing without a database, but it doesnt seem like you are using map as intended to begin with.

Seperate the logic of identifying the user in question by mapping to an assignment that will eventually populate the user if it exists and use that to pass into done().

ie
```
....
let match;
users.forEach((u) => {
  if(u.login === username && u.password === password) match = u;
});
...
// discern if match is valid and then resolve done() with whatever you are feeding it
// this looks like poc code so I wont nitpick
```
